core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -342,6 +344,21 @@ private SqlLibraryOperators() {
   public static final SqlOperator INFIX_CAST =
       new SqlCastOperator();
 
+  /** The "TO_NUMBER(string1, string2)" function; casts string1
+   * as hexadecimal to a NUMBER using the format specified in string2. */
+  @LibraryOperator(libraries = {TERADATA})
+  public static final SqlFunction TO_NUMBER =
+    new SqlFunction(
```

Comment:
Oracle[1] and PostgreSQL[2] also have TO_NUMBER function.

[1] https://docs.oracle.com/cd/B19306_01/server.102/b14200/functions191.htm
[2] http://www.postgresqltutorial.com/postgresql-to_number/

Reply:
Done the suggested changes.